internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/fastclient/grpc/VeniceGrpcEndtoEndTest.java

```diff
@@ -0,0 +1,157 @@
+    props.setProperty(CONTROLLER_AUTO_MATERIALIZE_META_SYSTEM_STORE, "true");
+    props.put(SERVER_HTTP2_INBOUND_ENABLED, "true");
+    props.put(SERVER_QUOTA_ENFORCEMENT_ENABLED, "true");
+    props.put(ENABLE_GRPC_READ_SERVER, "true");
+
+    cluster = ServiceFactory.getVeniceCluster(
+        new VeniceClusterCreateOptions.Builder().numberOfControllers(1)
+            .numberOfPartitions(1)
+            .numberOfRouters(1)
+            .numberOfServers(1)
+            .sslToStorageNodes(true)
+            .extraProperties(props)
+            .build());
+  }
+
+  @AfterClass
+  public void cleanUp() {
+    Utils.closeQuietlyWithErrorLogged(cluster);
+  }
+
+  public String writeData(String storeName) throws IOException {
+    // 1. Create a new store in Venice
+    String uniqueStoreName = Utils.getUniqueString(storeName);
+    NewStoreResponse response = cluster.getNewStore(uniqueStoreName);
+    UpdateStoreQueryParams params = new UpdateStoreQueryParams().setStorageQuotaInByte(Store.UNLIMITED_STORAGE_QUOTA);
+
+    Assert.assertEquals(response.getName(), uniqueStoreName);
+    Assert.assertNull(response.getError());
+
+    ControllerResponse updateStoreResponse = cluster.updateStore(uniqueStoreName, params);
+    Assert.assertNull(updateStoreResponse.getError());
+
+    // 2. Write Synthetic data to the store w/ writeSimpleAvroFileWithUserSchema
+    File inputDir = TestWriteUtils.getTempDataDirectory();
+    String inputDirPath = "file://" + inputDir.getAbsolutePath();
+    TestWriteUtils.writeSimpleAvroFileWithUserSchema(inputDir);
+
+    // 3. Run a push job to push the data to Venice (VPJ)
+    Properties vpjProps = TestWriteUtils.defaultVPJProps(cluster.getRandomRouterURL(), inputDirPath, uniqueStoreName);
+    TestWriteUtils.runPushJob("test push job", vpjProps);
+
+    cluster.useControllerClient(
+        controllerClient -> TestUtils.waitForNonDeterministicAssertion(60, TimeUnit.SECONDS, () -> {
+          StoreResponse storeResponse = controllerClient.getStore(uniqueStoreName);
+          Assert.assertEquals(storeResponse.getStore().getCurrentVersion(), 1);
+        }));
+
+    return uniqueStoreName;
+  }
+
+  @Test
+  public void testReadData() throws Exception {
+    String storeName = writeData("new-store");
+
+    // 4. Create thin client
+    AvroGenericStoreClient<Object, Object> avroClient = ClientFactory.getAndStartGenericAvroClient(
+        ClientConfig.defaultGenericClientConfig(storeName).setVeniceURL(cluster.getRandomRouterURL()));
+
+    // 4. Create fastClient
+    r2Client = ClientTestUtils.getR2Client(ClientTestUtils.FastClientHTTPVariant.HTTP_2_BASED_R2_CLIENT);
+    d2Client = D2TestUtils.getAndStartHttpsD2Client(cluster.getZk().getAddress());
+
+    com.linkedin.venice.fastclient.ClientConfig.ClientConfigBuilder<Object, Object, SpecificRecord> clientConfigBuilder =
+        new com.linkedin.venice.fastclient.ClientConfig.ClientConfigBuilder<>().setStoreName(storeName)
```

Comment:
Remove dependencies on`AbstractClientE2ESetup class

we can rewrite this block something like this:
```java


    ClientConfigBuilder<Object, Object, SpecificRecord> clientConfigBuilder =
        new ClientConfigBuilder<>().setStoreName(storeName)
            .setStoreMetadataFetchMode(StoreMetadataFetchMode.SERVER_BASED_METADATA)
            .setR2Client(ClientTestUtils.getR2Client(ClientTestUtils.FastClientHTTPVariant.HTTP_2_BASED_R2_CLIENT))
            .setD2Client(D2TestUtils.getAndStartHttpsD2Client(cluster.getZk().getAddress()))
            .setClusterDiscoveryD2Service(VeniceRouterWrapper.CLUSTER_DISCOVERY_D2_SERVICE_NAME)
            .setMaxAllowedKeyCntInBatchGetReq(150)
            .setMetadataRefreshIntervalInSeconds(5)
            .setMetricsRepository(new MetricsRepository())
            .setUseStreamingBatchGetAsDefault(true)
            .setSpeculativeQueryEnabled(false);

    AvroGenericStoreClient<String, GenericRecord> genericFastClient =
        ClientFactory.getAndStartGenericStoreClient(clientConfigBuilder.build());
        
        
```